### PR TITLE
Gnome 45 & 46 Port

### PR DESCRIPTION
**Pull request by petar-v**
Hi @lupantano ,

Thank you for the great extension! I use it on a daily basis and have updated it so that it is compliant with Gnome 45 and Gnome 46. 

This is as per https://gjs.guide/extensions/upgrading/gnome-shell-45.html.

Please test it locally. I am happy to address feedback and help get this published. :) 

Thanks!

Here is the built version:
[readingstrip@lupantano.gihthub.shell-extension.zip](https://github.com/lupantano/readingstrip/files/14622474/readingstrip%40lupantano.gihthub.shell-extension.zip)

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -15,198 +15,220 @@
...
+    this.focus_up.visible = this.strip_h.visible;
+    this.focus_down.visible = this.strip_h.visible;
+    this._settings.set_boolean("enabled", this.strip_h.visible);
+  }
+
+  enable() {
+    // Load settings
+    this._settings = this.getSettings();
+
+    this.icon_on = Gio.icon_new_for_string(
```
Comment (petar-v) [inline extension.js:88]: Could do a better state management job here so that it's more readable.

File: extension.js [hunk 2]
```diff
@@ -15,198 +15,220 @@
...
+    });
+    this._indicator.add_child(this.icon);
+    this._indicator.menu.addAction(_("Show/Hide"), () => this.toggleStrip());
+    this._indicator.menu.addAction(
+      _("Settings..."),
+      () => this.openPreferences(),
+      "settings-symbolic"
+    );
+
+    Main.panel.addToStatusArea(this.metadata.uuid, this._indicator);
```
Comment (petar-v) [inline extension.js:117]: It'd be good to split the UI building stuff into its own file/lib. This will make it easier to read.

**Comment by lupantano** [comment]
I thank you. I have made many changes to the code. I will import your work as soon as possible.

**Comment by lupantano** [comment]
I merged your code with the latest version of readingstrip.
It is definitely buggy, try it.

**Pull request closed without merging**